corehq/apps/change_feed/pillow.py

```diff
@@ -26,19 +27,20 @@ def __init__(self, kafka, data_source_type, data_source_name):
         self._data_source_name = data_source_name
 
     def process_change(self, pillow_instance, change, do_set_checkpoint=False):
-        document_type = _get_document_type(change.document)
-        if document_type:
-            assert change.document is not None
-            change_meta = ChangeMeta(
-                document_id=change.id,
+        try:
+            doc_meta = get_doc_meta_object_from_document(change.document)
+            change_meta = change_meta_from_doc_meta_and_document(
+                doc_meta=doc_meta,
+                document=change.document,
                 data_source_type=self._data_source_type,
                 data_source_name=self._data_source_name,
-                document_type=document_type,
-                document_subtype=_get_document_subtype(change.document),
-                domain=change.document.get('domain', None),
-                is_deletion=change.deleted,
+                doc_id=change.id,
             )
-            self._producer.send_change(get_topic(document_type), change_meta)
+        except MissingMetaInformationError:
+            pass
+        else:
+            assert not change.deleted
```

Comment:
👍 makes sense. Perhaps a comment in the Change & ChangeMeta class to
clarify the difference?